apps/client/src/translation/TranslationProvider.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useContext, useState } from 'react';
+
+import { langDe } from '@/translation/languages/de';
+import { langEn } from '@/translation/languages/en';
+
+const translationsList = {
+  en: langEn,
+  de: langDe,
+};
+const ALLOWED_LANGUAGES = Object.keys(translationsList);
+const DEFAULT_LANGUAGE = 'en';
+export const TranslationContext = createContext(undefined);
+
+export const TranslationProvider = ({ children }) => {
+  // Default language
+  const [language, setLanguageState] = useState('en');
+
+  const getString = (key, lang = language) => {
+    if (key in translationsList[lang]) {
+      return translationsList[lang][key];
+    } else if (lang !== DEFAULT_LANGUAGE) {
+      /* If the key does not exist in the chosen language, try to load it in the default language. */
+      return getString(key, 'en');
+    } else {
+      /* We are here if the key does not exist in the default language. */
+      return undefined;
+    }
+  };
+
+  const setLanguage = (language) => {
+    language = language.toLowerCase();
+    if (ALLOWED_LANGUAGES.includes(language)) {
+      setLanguageState(language);
+      console.info(`Language set to ${language}`);
+    } else {
+      console.warn(`Language code ${language} does not exist.`);
+    }
+  };
+
+  const contextValue = {
+    getString,
+    setLanguage,
+  };
+
+  return <TranslationContext.Provider value={contextValue}>{children}</TranslationContext.Provider>;
+};
+
+export const useTranslation = () => {
+  const { getString, setLanguage } = useContext(TranslationContext);
+  return { getString, setLanguage };
```

Comment:
I wonder if we could use something less generic like `getLocalisedString`
What do you think?

Reply:
Yes, sounds like a good thing to do.